libsolidity/codegen/ir/WellKnownUtilFunctions.cpp

```diff
@@ -0,0 +1,260 @@
+		return Whiskers(R"(
+			function m_allocateUnbounded() -> mobj {
+				mobj := mload(<freeMemoryPointer>)
+			}
+		)")
+		("freeMemoryPointer", to_string(CompilerUtils::freeMemoryPointer))
+		.render();
+	});
+
+	_functionCollector.createFunction("m_finalize", [&]() {
+		return Whiskers(R"(
+			function m_finalize(mobj, size) {
+				size := <roundUp>(size)
+				let newFreePtr := add(mobj, size)
+				// protect against overflow
+				if or(gt(size, 0xffffffffffffffff), lt(newFreePtr, mobj)) { <panic>() }
+				mstore(<freeMemoryPointer>, newFreePtr)
+			}
+		)")
+		("freeMemoryPointer", to_string(CompilerUtils::freeMemoryPointer))
+		("panic", utils.panicFunction())
+		("roundUp", utils.roundUpFunction())
+		.render();
+	});
+
+	_functionCollector.createFunction("m_discard", [&]() {
+		return "\nfunction m_discard(mobj) { }\n";
+	});
+
+	_functionCollector.createFunction("m_load", [&]() {
+		return R"(
+			function m_load(mobj, offset) -> value {
+				value := mload(m_absolute(mobj, offset))
+			}
+		)";
+	});
+
+	_functionCollector.createFunction("m_store", [&]() {
+		return R"(
+			function m_store(mobj, offset, value) {
+				mstore(m_absolute(mobj, offset), value)
+			}
+		)";
+	});
+
+	_functionCollector.createFunction("m_store8", [&]() {
+		return R"(
+			function m_store8(mobj, offset, value) {
+				mstore8(m_absolute(mobj, offset), value)
+			}
+		)";
+	});
+
+	_functionCollector.createFunction("m_slice", [&]() {
+		return R"(
+			function m_slice(mobj, offset) -> ret {
+				ret := add(mobj, offset)
+			}
+		)";
+	});
+
+	_functionCollector.createFunction("m_calldatacopy", [&]() {
+		return R"(
+			function m_calldatacopy(mobj, from_offset, size) {
+				calldatacopy(m_absolute(mobj, 0), from_offset, size)
+			}
+		)";
+	});
+
+	_functionCollector.createFunction("m_codecopy", [&]() {
+		return R"(
+			function m_codecopy(mobj, from_offset, size) {
+				codecopy(m_absolute(mobj, 0), from_offset, size)
+			}
+		)";
+	});
+
+	_functionCollector.createFunction("m_extcodecopy", [&]() {
+		return R"(
+			function m_extcodecopy(addr, mobj, from_offset, size) {
+				extcodecopy(addr, m_absolute(mobj, 0), from_offset, size)
+			}
+		)";
+	});
+
+	_functionCollector.createFunction("m_returndatacopy", [&]() {
+		return R"(
+			function m_returndatacopy(mobj, from_offset, size) {
+				returndatacopy(m_absolute(mobj, 0), from_offset, size)
+			}
+		)";
+	});
+
+	_functionCollector.createFunction("m_create", [&]() {
+		return R"(
+			function m_create(value, mobj, size) {
+				create(value, m_absolute(mobj, 0), size)
+			}
+		)";
+	});
+
+	_functionCollector.createFunction("m_create2", [&]() {
+		return R"(
+			function m_create2(value, mobj, size, salt) {
+				create2(value, m_absolute(mobj, 0), size, salt)
+			}
+		)";
+	});
+	_functionCollector.createFunction("m_call", [&]() {
+		return R"(
+			function m_call(gasAllowance, addr, value, mobjInput, inputSize, mobjOutput, outputSize) -> success {
```

Comment:
Hm... since we do have ``m_slice``, it would make sense to drop the sizes here - but that would only work, if we stored the size in the handle, right? Might be another reason for actually storing the sizes.